jwst/background/background_sub.py

```diff
@@ -39,53 +49,86 @@ def background_sub(input_model, bkg_list, sigma, maxiters):
 
     # Subtract the average background from the member
     log.debug(' subtracting avg bkg from {}'.format(input_model.meta.filename))
+
     result = subtract_images.subtract(input_model, bkg_model)
 
-    # We're done. Return the result.
+    # We're done. Return the background model and the result.
     return bkg_model, result
 
 
 def average_background(bkg_list, sigma, maxiters):
     """
-    Average multiple background exposures into a combined data model
+    Average multiple background exposures into a combined data model.
+    Processes either 2D (rate) or 3D (rateints) backgrounds.
 
     Parameters:
     -----------
-
     bkg_list: filename list
         List of background exposure file names
 
+    sigma: float, optional
+        Number of standard deviations to use for both the lower
+        and upper clipping limits.
+
+    maxiters: int or None, optional
+        Maximum number of sigma-clipping iterations to perform
+
     Returns:
     --------
-
     avg_bkg: data model
         The averaged background exposure
-
     """
 
+    # Determine the type and dimensionality of the background files
+    try:
```

Comment:
The input background exposures can be more than just `ImageModel` and `CubeModel`. IFU exposures, for example, will be `IFUImageModel`. So I think it's best to not try to get them to conform to any particular datamodel and just figure out the dimensions using
```
bkg_model = datamodels.open(bkg_list[0])
image_shape = bkg_model.data.shape
bkg_dim = len(image_shape)
```
bkg_dim will have values of either 2 or 3.

Reply:
So this logic can be collapsed to just those 3 lines of code suggested above and `image_shape` is still available for use below, without the need for separate try/except blocks.